service/storage_proxy.cc

```diff
@@ -4403,7 +4403,14 @@ future<result<>> storage_proxy::schedule_repair(locator::effective_replication_m
     if (diffs.empty()) {
         return make_ready_future<result<>>(bo::success());
     }
-    return mutate_internal(diffs | std::views::values | std::views::transform([ermp] (auto& v) { return read_repair_mutation{std::move(v), ermp}; }), cl, false, std::move(trace_state), std::move(permit));
+    return mutate_internal(
+            diffs |
+                    std::views::values |
+                    std::views::transform([ermp] (auto& v) { return read_repair_mutation{std::move(v), ermp}; }) |
+                    // mutate_internal() is asynchronous, so we have to materialize this view,
```

Comment:
the comment is not accurate. the problem is not that `mutate_internal()` is asynchronous. the root cause is that the range view mutates the underlying container. so every time, we iterate through the range view, we consume the values in `diffs`, as explained in the cover letter of the pull request.

could you please update the comment accordingly?

Reply:
fixed in v3